app/components/email-list/component.js

```diff
@@ -0,0 +1,57 @@
+import Ember from "ember";
+const { set, computed, getProperties, setProperties, get } = Ember;
+
+export default Ember.Component.extend({
+  emailDetails: false,
+  emailWrapperClass: "",
+  toggleAllLabel: true,
+  init() {
+    this._super(...arguments);
+    let settings = JSON.parse(localStorage.getItem("settings"));
+    let { emailDetails } = getProperties(this, "emailDetails");
+    setProperties(this, {
+      emailDetails: settings !== null ? settings.detailsPage : emailDetails,
+      emailWrapperClass:
+        settings && settings.detailsPage == true
+          ? "email-container email-details-disabled"
+          : "email-container",
+    });
+  },
+  markAllItemsRead: computed("emailList.@each.read", {
+    get() {
+      return get(this, "emailList").isEvery("read");
+    },
+  }),
+  defaultEmailDetails: computed("emailList.@each.[]", {
+    get() {
+      return get(this, "emailList").filterBy("id", "1");
+    },
+  }),
+  actions: {
+    getEmailDetails(item) {
+      let selectedEmailDetails = get(this, "emailList").filterBy("id", item.id);
+      set(item, "read", true);
+      set(this, "defaultEmailDetails", selectedEmailDetails);
+    },
+    toggleAll() {
+      let { emailList, markAllItemsRead, toggleAllLabel } = getProperties(
```

Comment:
logic here should be,
1. lets create a boolean variable isRead: false
2. create a computed property toggleAllLabel -> dependent on isRead boolean variable and return appropriate "Mark as read" or "Mark as unread".
3. create an action toggleEmails()  -> run the loop of all emails and set a toggled value of isRead and finally toggle isRead value


u will get the state of toggleAllLabel which will either be "read" or "unread"